pandas/tests/test_strings.py

```diff
@@ -26,6 +29,157 @@ def assert_series_or_index_equal(left, right):
+    'title', 'swapcase',
+    'isalpha', 'isnumeric', 'isalnum',
+    'isdigit', 'isdecimal', 'isspace',
+    'islower', 'isupper', 'istitle'
+], [tuple()] * 100))
+ids, _ = zip(*_all_string_methods)  # use method name as fixture-id
+
+
+@pytest.fixture(params=_all_string_methods, ids=ids)
+def all_string_methods(request):
+    """
+    Fixture for all public methods of `StringMethods`
+
+    This fixture returns a tuple of the method name and a list of sample values
+    for the required positional arguments of that method.
+    """
+    return request.param
+
+
+_all_allowed_skipna_inferred_dtypes = [
+    ('string', ['a', np.nan, 'c']),
+    ('unicode' if not PY3 else 'string', [u('a'), np.nan, u('c')]),
+    ('bytes' if PY3 else 'string', [b'a', np.nan, b'c']),
+    ('empty', [np.nan, np.nan, np.nan]),
+    ('empty', []),
+    ('mixed-integer', ['a', np.nan, 2]),
+    ('mixed', ['a', np.nan, 2.0])]
+ids, _ = zip(*_all_allowed_skipna_inferred_dtypes)  # use inferred type as id
+
+
+@pytest.fixture(params=_all_allowed_skipna_inferred_dtypes, ids=ids)
+def all_allowed_skipna_inferred_dtypes(request):
+    """
+    Fixture for all (inferred) dtypes allowed in StringMethods.__init__
+
+    Returns an np.ndarray that will be inferred to have the given dtype (when
+    skipping missing values).
+
+    The allowed (inferred) types are:
+    * 'string'
+    * 'unicode' (if PY2)
+    * 'empty'
+    * 'bytes' (if PY3)
+    * 'mixed'
+    * 'mixed-integer'
+    """
+    inferred_dtype, values = request.param
+    values = np.array(values, dtype=object)  # object dtype to avoid casting
+
+    # make sure the inferred dtype of the fixture is as requested
+    assert inferred_dtype == lib.infer_dtype(values, skipna=True)
+
+    return inferred_dtype, values
+
+
+# categoricals are handled separately
+_all_skipna_inferred_dtypes = _all_allowed_skipna_inferred_dtypes + [
+    ('floating', [1.0, np.nan, 2.0]),
+    ('integer', [1, np.nan, 2]),
+    ('mixed-integer-float', [1, np.nan, 2.0]),
+    ('decimal', [Decimal(1), np.nan, Decimal(2)]),
+    ('boolean', [True, np.nan, False]),
+    ('datetime64', [np.datetime64('2013-01-01'), np.nan,
+                    np.datetime64('2018-01-01')]),
+    ('datetime', [Timestamp('20130101'), np.nan, Timestamp('20180101')]),
+    ('date', [date(2013, 1, 1), np.nan, date(2018, 1, 1)]),
+    # The following two dtypes are commented out due to GH 23554
+    # ('complex', [1 + 1j, np.nan, 2 + 2j]),
+    # ('timedelta64', [np.timedelta64(1, 'D'),
+    #                  np.nan, np.timedelta64(2, 'D')]),
+    ('timedelta', [timedelta(1), np.nan, timedelta(2)]),
+    ('time', [time(1), np.nan, time(2)]),
+    ('period', [Period(2013), NaT, Period(2018)]),
+    ('interval', [Interval(0, 1), np.nan, Interval(0, 2)])]
+ids, _ = zip(*_all_skipna_inferred_dtypes)  # use inferred type as fixture-id
+
+
+@pytest.fixture(params=_all_skipna_inferred_dtypes, ids=ids)
+def all_skipna_inferred_dtypes(request):
+    """
```

Comment:
so makes sense to also use these in the inferred_type tests then.

Reply:
You mean for the tests in `tests/dtypes/test_inference.py`?

They're being used in this module already.

Reply:
not sure what you mean

Reply:
I'm asking if you want me to move this particular fixture to `pandas/conftest.py` and then test it within the dtype tests (because this is effectively a dtype thing).

Reply:
yes